pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -105,39 +109,67 @@ type streamIngestionResumer struct {
 	job *jobs.Job
 }
 
-func ingest(
-	ctx context.Context,
-	execCtx sql.JobExecContext,
-	streamAddress streamingccl.StreamAddress,
-	oldTenantID roachpb.TenantID,
-	newTenantID roachpb.TenantID,
-	streamID streaming.StreamID,
-	startTime hlc.Timestamp,
-	progress jobspb.Progress,
-	ingestionJobID jobspb.JobID,
-) error {
+func ingest(ctx context.Context, execCtx sql.JobExecContext, ingestionJob *jobs.Job) error {
+	details := ingestionJob.Details().(jobspb.StreamIngestionDetails)
+	progress := ingestionJob.Progress()
+	streamAddress := streamingccl.StreamAddress(details.StreamAddress)
+
+	startTime := progress.GetStreamIngest().StartTime
+	// Start from the last checkpoint if it exists.
+	if h := progress.GetHighWater(); h != nil && !h.IsEmpty() {
+		startTime = *h
+	}
+
+	// If there is an existing stream ID, reconnect to it.
+	streamID := streaming.StreamID(details.StreamID)
 	// Initialize a stream client and resolve topology.
 	client, err := streamclient.NewStreamClient(streamAddress)
 	if err != nil {
 		return err
 	}
 	ingestWithClient := func() error {
-		// TODO(dt): if there is an existing stream ID, reconnect to it.
+		ro := retry.Options{
+			InitialBackoff: 1 * time.Second,
+			Multiplier:     2,
+			MaxBackoff:     10 * time.Second,
+			MaxRetries:     5,
+		}
+		// Make sure the producer job is active before start the stream replication.
+		var status streampb.StreamReplicationStatus
+		for r := retry.Start(ro); r.Next(); {
+			status, err = client.Heartbeat(ctx, streamID, startTime)
```

Comment:
I thought that the ingestion processor was supposed to heartbeat src cluster periodically; I only see this happen here... where do we periodically ping src cluster?